bridge/src/sqlite3-exchange-history-store.ts

```diff
@@ -0,0 +1,78 @@
+import { ExchangeHistory, IExchangeHistoryStore } from "./interfaces/exchange-history-store";
+import { Database } from "sqlite3";
+import { promisify } from "util";
+
+export class Sqlite3ExchangeHistoryStore implements IExchangeHistoryStore {
+    private readonly _database: Database;
+    private closed: boolean;
+
+    private constructor(database: Database) {
+        this._database = database;
+        this.closed = false;
+    }
+    put(history: ExchangeHistory): Promise<void> {
+        this.checkClosed();
+
+        const {
+            network,
+            tx_id,
+            sender,
+            recipient,
+            amount,
+            timestamp,
+        } = history;
+
+        const run: (sql: string, params: any[]) => Promise<void> = promisify(this._database.run.bind(this._database));
+        return run(
+            "INSERT INTO exchange_histories(network, tx_id, sender, recipient, amount, timestamp) VALUES (?, ?, ?, ?, ?, ?)",
+            [network, tx_id, sender, recipient, amount, timestamp]);
+    }
+
+    async transferredAmountInLast24Hours(network: string, sender: string): Promise<number> {
+        this.checkClosed();
+        const get: (sql: string, params: any[]) => Promise<{ "total_amount": string | null } > = promisify(this._database.get.bind(this._database));
+        const row = await get("SELECT SUM(amount) as total_amount FROM exchange_histories WHERE network = ? AND sender = ? AND datetime(timestamp) > datetime('now', '-1 day');", [network, sender]);
+
+        return parseInt(row["total_amount"] ?? "0");
+    }
+
+    static async open(path: string): Promise<Sqlite3ExchangeHistoryStore> {
+        const database = new Database(path);
+        await this.initialize(database);
+        return new Sqlite3ExchangeHistoryStore(database);
+    }
+
+    private static async initialize(database: Database): Promise<void> {
+        const CREATE_TABLE_QUERY = `CREATE TABLE IF NOT EXISTS exchange_histories (
+            network TEXT NOT NULL,
+            tx_id TEXT NOT NULL,
+            sender TEXT NOT NULL,
```

Comment:
How about adding an index to `sender` for querying.

Reply:
I tried to add the index in 97f35ce commit. Please review again.